client/app/components/faq/faqlist.js

```diff
@@ -35,94 +30,97 @@ class FaqList extends Component {
 
 	componentDidMount() {
 		this.fetchData()
+			// eslint-disable-next-line no-console
 			.then(() => console.log("Finished mounting!"))
+			// eslint-disable-next-line no-console
 			.catch(err => console.error(err));
 	}
 
 	componentWillReceiveProps(nextProps) {
 		// Avoiding refresh if possible
-	  if (nextProps.currentQuestions !== this.state.currentQuestions) {
-		 this.setState({ currentQuestions: nextProps.currentQuestions });
-	  }
+		if (nextProps.currentQuestions !== this.state.currentQuestions) {
+			this.setState({ currentQuestions: nextProps.currentQuestions });
+		}
 	}
 
-	render() {
-		return (
-			<ScrollView
-				style={FaqStyles.faqPadding, {backgroundColor: 'white'}}
-				refreshControl={
-					<RefreshControl
-						refreshing={this.state.isRefreshing}
-						onRefresh={this._onRefresh.bind(this)}
-					/>
-				}
-			>
-				<FlatList
-					data={this.state.currentQuestions}
-					renderItem={this._renderListItem}
-					extraData={this.state}
-					keyExtractor={this._keyExtractor}
-					ListHeaderComponent={this._renderHeader}
-				/>
-			</ScrollView>
-		);
+	onRefresh = () => {
+		this.setState({isRefreshing: true});
+		this.fetchData().then(() => {
+			this.setState({isRefreshing: false});
+		});
 	}
 
 	fetchData() {
 		return fetch(`${API_URL}/list`)
 			.then(response => response.json())
 			.then(faqList => this.props.onFAQLoaded(faqList))
+			// eslint-disable-next-line no-console
 			.catch(err => console.error(err));
 	}
 
-	_onRefresh() {
-		this.setState({isRefreshing: true});
-		this.fetchData().then(() => {
-			this.setState({isRefreshing: false});
-		});
-	}
-
-	_renderListItem = ({ item, index }) => {
+	renderListItem({ item, index }) {
 		let rowBg = index % 2 == 1 ? FaqStyles.faqListItemOdd : null;
+		
+		const icon = (
+			<Icon
+				name="chevron-thin-right"
+				size={30}
+				color="#787878"
+				style={{marginTop: 5}}
+			/>
+		);
+
 		return (
 			<ListItem
-				containerStyle={FaqStyles.faqListItem, rowBg}
+				containerStyle={{...FaqStyles.faqListItem, rowBg}}
 				titleStyle={FaqStyles.faqListItemText}
 				key={item.id}
 				title={item.question}
-				onPress={() => this._renderFaqDetails(item, index)}
-				rightIcon={<Icon
-								name={'chevron-thin-right'}
-								size={30}
-								color={'#787878'}
-								style={{marginTop: 5}}
-						  />}
+				onPress={this.renderFaqDetails(item, index)}
+				rightIcon={icon}
 			/>
 		);
 	}
 
-	_renderHeader = () => {
+	renderHeader = () => {
 		//TODO: Add actual search
 		return <SearchBar placeholder="Search for questions here!" lightTheme />;
 	}
 
-	_renderFaqDetails = (question, index) => {
-		this.props.navigation.navigate('FaqDetails', {
-			index: index,
-			currentQuestion: question,
-			questionList: this.state.currentQuestions,
-		});
+	renderFaqDetails = (question, index) => {
```

Comment:
Had to implement this as a higher order function, because [of this](https://stackoverflow.com/questions/36677733/why-shouldnt-jsx-props-use-arrow-functions-or-bind)

Reply:
You're right that this shouldn't have been implemented as an arrow function passed to `onPress`, but I would consider reimplementing this by binding the two variables within the `renderListItem` function. i.e.
`this.renderFaqDetails.bind(this, question, index)`

Even though it might be less obvious what `bind` is doing, but the return value of the function makes more sense than returning a function.

Reply:
Binding functions in jsx decreases performance a lot. This is as every `onPress` call creates a new instance of the binded function in memory, which can add up a lot over time for common processes. I think returning a function is a fine pattern for now and it saves a lot of performance in the long term. Thoughts?  

In addition, I'm looking into memoization so that we can further improve performance on these functions.

Reply:
I still don't quite like how it looks, but it's fine.  Maybe add in a comment explaining why it's implemented as such.